src/main/java/org/opensearch/neuralsearch/query/NeuralKNNQueryBuilder.java

```diff
@@ -0,0 +1,158 @@
+ * 2. Allow extension with neural-search-specific information (e.g., query text)
+ */
+public class NeuralKNNQueryBuilder extends AbstractQueryBuilder<NeuralKNNQueryBuilder> {
+    private final KNNQueryBuilder knnQueryBuilder;
+
+    /**
+     * Wraps KNN Lucene query to support neural search extensions.
+     * Delegates core operations to the underlying KNN query.
+     */
+    public static class NeuralKNNQuery extends Query {
+        private final Query knnQuery;
+
+        public NeuralKNNQuery(Query knnQuery) {
+            this.knnQuery = knnQuery;
+        }
+
+        @Override
+        public String toString(String field) {
+            return knnQuery.toString(field);
+        }
+
+        @Override
+        public void visit(QueryVisitor visitor) {
+            // Delegate the visitor to the underlying KNN query
+            knnQuery.visit(visitor);
+        }
+
+        @Override
+        public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+            // Delegate weight creation to the underlying KNN query
+            return knnQuery.createWeight(searcher, scoreMode, boost);
+        }
+
+        @Override
+        public Query rewrite(IndexReader reader) throws IOException {
+            Query rewritten = knnQuery.rewrite(reader);
+            if (rewritten == knnQuery) {
+                return this;
+            }
+            return new NeuralKNNQuery(rewritten);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (this == other) return true;
+            if (other == null || getClass() != other.getClass()) return false;
+            NeuralKNNQuery that = (NeuralKNNQuery) other;
+            return Objects.equals(knnQuery, that.knnQuery);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(knnQuery);
+        }
+    }
+
+    /**
+     * Factory method to create NeuralKNNQueryBuilder with given parameters.
+     * Centralizes KNNQueryBuilder creation to handle breaking changes.
+     */
+    public static NeuralKNNQueryBuilder fromKNNBuilder(
+        String fieldName,
+        float[] vector,
+        Integer k,
+        QueryBuilder filter,
+        Float maxDistance,
+        Float minScore,
+        Boolean expandNested,
+        Map<String, ?> methodParameters,
+        RescoreContext rescoreContext
+    ) {
+        KNNQueryBuilder knnBuilder = KNNQueryBuilder.builder()
+            .fieldName(fieldName)
+            .vector(vector)
+            .k(k)
+            .filter(filter)
+            .maxDistance(maxDistance)
+            .minScore(minScore)
+            .expandNested(expandNested)
+            .methodParameters(methodParameters)
+            .rescoreContext(rescoreContext)
+            .build();
+        return new NeuralKNNQueryBuilder(knnBuilder);
+    }
+
+    private NeuralKNNQueryBuilder(KNNQueryBuilder knnQueryBuilder) {
+        super();
```

Comment:
You don't need to call `super()`. It will be called automatically.

Reply:
ack

Reply:
Removed.